model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+import org.openremote.model.value.*;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.openremote.model.value.MetaItemType.*;
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "reported"
+})
+public class State {
+
+//    public ReportedState reportedState;
+    @JsonProperty("reported")
+    private Map<String, Object> reported;
+
+
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(State.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
+        sb.append("reported");
+        sb.append('=');
+        sb.append(((this.reported == null) ? "<null>" : this.reported));
+        sb.append(',');
+        if (sb.charAt((sb.length() - 1)) == ',') {
+            sb.setCharAt((sb.length() - 1), ']');
+        } else {
+            sb.append(']');
+        }
+        return sb.toString();
+    }
+
+
+    //TODO: Add real timestamp reported by device, check issue #1156
+    //TODO: Add description of each attribute (No idea where to put that yet)
+    //TODO: Move any function from here; it's a POJO.
+    //TODO: Figure out what the parameters: pr, alt, ang, sat, sp, and evt do
+    // and implement their functionality (I can guess but I want to know exactly).
+    public AttributeMap GetAttributes(Map<Integer, TeltonikaParameter> params, AttributeMap additionalAttributes, Logger logger){
+        AttributeMap attributes = new AttributeMap();
+        for (Map.Entry<String,Object> entry : reported.entrySet()){
+            //Check if parameter is a number value, pointing to a TeltonikaParameter
+            String entryId = entry.getKey();
+            int parsedEntryId;
+            try {
+                parsedEntryId = Integer.parseInt(entryId);
+            }catch (Exception e){
+                continue;
+            }
+            //If the Teltonika Parameter HashMap doesn't contain the requested AVL ID, continue:
+
+            if (!params.containsKey(parsedEntryId)) continue;
+
+            //Retrieve the parameter data
+            TeltonikaParameter parameter = params.get(Integer.valueOf(entry.getKey()));
+
+            //Create the MetaItem Map
+            MetaMap metaMap = new MetaMap();
+
+            // Figure out its attributeType
+            ValueDescriptor<?> attributeType = GetAttributeType(parameter);
+
+            //Retrieve its coerced value
+            Optional<?> value;
+            try {
+                //Inner method returns Optional.empty, but still throws and prints exception. A lot of clutter, but the exception is handled.
+                value = ValueUtil.getValueCoerced(entry.getValue(), attributeType.getType());
+                if (value.isEmpty()){
+                    attributeType = ValueType.TEXT;
```

Comment:
This could try to change an existing attributes type which seems odd

Reply:
How do you think this should be handled? Should I not process this attribute or change its type?